data_dump/README.md

```diff
@@ -0,0 +1,152 @@
+vector at the given `REPORT_STEP` indices. Each NetCDF variable is a separate
+`GEN_DATA` entry.
+
+## `SUMMARY`
+
+All `SUMMARY`s are in the same group under `/REAL_{index}/SUMMARY`. This dataset
+contains the dimension `report_step`, which is the `time-map`. Each NetCDF
+variable is a separate `SUMMARY` entry.
+
+## `FIELD`
+
+Each ERT `FIELD` is a subgroup of `/REAL_{index}/FIELD`, where the name of the
+group is the name of the field. The dimensions are `x`, `y` and `z`, which
+correspond to the 3-D axes of fields. The data for the field is given by the
+`VALUES` variable.
+
+## `SURFACE`
+
+Each ERT `SURFACE` is a subgroup of `/REAL_{index}/SURFACE`, where the name of
+the group is the name of the surface. The data for the surface is given by the
+`VALUES` variable.
+
+# Reading
+
+To programmatically read data from the files it is recommended to use both the
+`netCDF4` package for low-level access, and `xarray` package for manipulating or
+converting the data.
+
+The `xarray` package does not make it easy to navigate groups which is why
+`netCDF4` is used. `xarray` uses `netCDF4` as a backend to open NetCDF version 4
+files, so both packages need to be installed anyway.
+
+## `netCDF4` package
+
+Groups can be navigated using `netCDF4`. First, open a file:
+
+``` python
+import netCDF4
+
+ds = netCDF4.Dataset("some-dataset.nc")
+```
+
+Groups are given using the `.groups` property. This returns a Python `dict` with
+the keys being the name of the group, and the value being a `netCDF4.Group`
+object. Groups may have subgroups.
+
+``` python
+for name, group in ds.groups.items():
+    for subname, subgroup in group.group.items():
+        print(f"{name}/{subname} = {subgroup}")
+```
+
+Specific subgroups can be accessed by `ds.groups["some_group"]` or
+`ds.group["some_group"]`. It is also possible to access subgroups by path, as if
+in a file structure: `ds["/foo/bar/baz"]`, which is equivalent to
+`ds.groups["foo"].groups["bar"].groups["baz"]`.
+
+NetCDF datasets have _variables_, which store actual data, and _dimensions_ (aka
+_coordinates_), which are the axes for the data. Within a group, if two
+different variables refer to a dimension by one name, then both variables will
+have the same dimension values. For example, the variables for porosity and
+pressure are different, but may share the same `x`, `y`, `z` axes.
+
+To access dimensions, use the `.dimensions` property. To access variables, use
+the `.variables` property. Both return Python `dict`s with keys as the dimension
+or variable names and the values are `netCDF4` data class instances. Note that
+dimensions are also variables. Unlike `xarray`, the low-level `netCDF4` package
+has no concept of "data variable". To get the data variables, do:
+
+``` python
+data_vars = {
+  key: val
+  for key, val in ds.variables.items()
+  if key not in ds.dimensions
+}
+```
+
+`netCDF4` variables are iterable their data can be obtained by calling
```

Comment:
Don't understand this I think.

If I do this:

`ds["/REAL_0/FIELD/FIELD_1"].variables`

I get the dict

{'VALUES': <class 'netCDF4._netCDF4.Variable'>
 float64 VALUES(x, y, z)
     _FillValue: nan
 path = /REAL_0/FIELD/FIELD_1
 unlimited dimensions: 
 current shape = (4, 4, 1)
 filling on}

so list of that would give:

['VALUES']

What do you mean that variables are iterable?

Reply:
I don't use the `.variables` property. In your example, it's possible to do:

```py
list(ds["/REAL_0/FIELD/FIELD_1"]["VALUES"])
```